explainer.md

```diff
@@ -431,130 +409,90 @@ should be at the discretion of the user agent. For example, the site shouldn't
+* `navigator.clearClientBadge()`: Clears the badge for the current
+  document.
+* `navigator.setAppBadge([contents])`: Sets the badge for the matching app(s) to
+  *contents* (an integer), or to "flag" if *contents* is omitted. If *contents*
+  is 0, clears the badge for the matching app(s).
+* `navigator.clearAppBadge()`: Clears the badge for the matching app(s).
+
+The **matching app(s)** has a different meaning depending on the context:
+
+* If called from a document, this refers to an installed app that this document
+  is [within
+  scope](https://www.w3.org/TR/appmanifest/#dfn-within-scope-manifest) of. If
+  multiple apps match, it is the one with the most specific scope. If none
+  match, the method has no effect. (Zero or one apps.)
+* If called from a service worker, this refers to all apps whose scope URL
+  [matches the service worker
+  registration](https://www.w3.org/TR/service-workers-1/#scope-match-algorithm)
+  of this service worker. (Zero or more apps.)
 
 **Note**: Should we have a separate overload for boolean flags now, as discussed in [Issue 19](https://github.com/WICG/badging/issues/19) and [Issue 42](https://github.com/WICG/badging/issues/42)?
 
 ### UX treatment
-Badges may appear in any place that the user agent deems appropriate. In general, these places should be obviously related to the pages being badged, so users understand what the status is for. Appropriate places could include:
+Badges may appear in any place that the user agent deems appropriate. In general, these places should be obviously related to the pages being badged, so users understand what the status is for.
+
+Appropriate places for a document badge could include:
+
 - Tab favicons.
-- Bookmark icons.
-- A "most visited sites" menu, e.g., on the user agent's "new tab" page.
-- [OS Specific Contexts](#OS-Specific-Contexts) for [Installed Web Applications](https://www.w3.org/TR/appmanifest/#installable-web-applications).
+- OS-specific contexts for app window icons if the document is open in a [standalone window](https://www.w3.org/TR/appmanifest/#display-modes).
 
-**Note**: When showing a badge in an [OS Specific Context](#OS-Specific-Contexts) user agents should attempt reuse existing [operating system APIs and conventions](docs/implementation.md), to achieve a native look-and-feel.
+App badges are shown in OS-specific contexts. User agents should attempt reuse existing [operating system APIs and conventions](docs/implementation.md), to achieve a native look-and-feel for the badge.
 
 ## Security and Privacy Considerations
-The API is set only, so data badged can't be used to track a user. Whether the API is present could possibly be used as a bit of entropy to fingerprint users, but this is the case for all new APIs.
+The API is write-only, so data badged can't be used to track a user. Whether the API is present could possibly be used as a bit of entropy to fingerprint users, but this is the case for all new APIs.
```

Comment:
We should probably mention that it's only exposed in secure contexts and not available to third-party iframe unless explicitly granted by feature policy.

Reply:
Secure contexts: Done

Third-party iframe: We actually do allow this, but if you call the method from an iframe, you set the badge on the iframed contents' origin, not the containing page. This seems reasonable to me.

Reply:
> Third-party iframe: We actually do allow this, but if you call the method from an iframe, you set the badge on the iframed contents' origin, not the containing page. This seems reasonable to me.

heh... I kinda like that :) We might need to keep some kind of open issue around this as I have a feeling it's going to come up for discussion. I think the main pushback we will get from this API is that it risks high user annoyance, so we need to have a good story in place for third-party iframes.